### PR TITLE
Ensure implicit none via compiler option

### LINKED ISSUE #63: Compilation with optimalization points to use of uninitialized variables
We should check whether the Warnings are valid (at least those in `transform.F90` seems to be.

### LINKED ISSUE #76: Use `--implicit-none` compiler option
This option should be a default in the Makefile or in the configure - > make.vars.
This would mean we will automatically catch any code that uses implicit typing in Github Actions in the future.

### PR DESCRIPTION
**Pull request by danielhollas**
Adding `-fimplicit-none` to FFLAGS ensures that we always have `implicit none` in our modules and functions.
I fixed one missing `implicit none` (caught an unitialized variable!) and converted one last implicit typing in `random.F90`.

l also moved `mod_const` from `modules.F90` to a separate file `constants.F90` to break circular dependency due to the 'DP' constant. This allowed me reorder compilation of a some files and make 'fatal_error()' accessible to all modules. In general, our goal should be to have one module per file, so while I was in there, I also did the same for `mod_files`.

Other small changes were done to squash compiler warnings. I also did a bit of cleanup in `mass_init` and wrote some unit tests to verify this functionality (especially user defined masses code was messy and not fully tested).

Closes #76, Closes #63

### CODE REVIEW

File: src/files.F90 [hunk 1]
```diff
@@ -0,0 +1,207 @@
+! Module for permanent file handling
```
Comment (danielhollas) [inline src/files.F90:1]: No changes here, just moved from module.F90

File: src/random.F90 [hunk 2]
```diff
@@ -542,8 +542,11 @@ end subroutine xuwarm
 
 
       FUNCTION R1MACH()
-      IMPLICIT real(DP) (A-H,O-Z)
-      PARAMETER (ONE=1.D0,TWO=2.D0,HALF=0.5D0)
+      real(DP), parameter :: ONE=1.D0, TWO=2.D0, HALF=0.5D0
```
Comment (danielhollas) [inline src/random.F90:545]: Last `implicit real` :-)

File: src/constants.F90 [hunk 3]
```diff
@@ -0,0 +1,38 @@
+! Exact values for ANG,AUTOKCAL and AUTODEBYE,me,AUTOM,AUTOFS,AMU:
```
Comment (danielhollas) [inline src/constants.F90:1]: No changes, just moved from `modules.F90`

File: src/modules.F90 [hunk 4]
```diff
@@ -120,29 +83,7 @@ module mod_system
    ! cannot use this
 !   use mod_utils, only: abinerror
    implicit none
-   ! TODO: move more stuff into Properties derived type
-   ! https://kiwi.atmos.colostate.edu/fortran/docs/fortran2012.key-8.pdf
-   ! TODO: We should probably move this outside of the modules.f90
-
-   ! Extra electronic properties (not mandatory)
```
Comment (danielhollas) [inline src/modules.F90:127]: These were just stubs, not used anywhere, so I removed them for now.

File: src/modules.F90 [hunk 5]
```diff
@@ -372,24 +289,58 @@ subroutine mass_init(masses, massnames)
...
-            print*,'I hope you specified the mass in namelist "system"'
+            if (my_rank == 0) then
+               print*,'Atom name ', names(i), ' was not found in the library.'
+               print*,'Using user-defined mass from input file'
+            end if
          end select
       end do
 
-      call init_usermass()
+      ! Check for duplicate user defined atom names
```
Comment (danielhollas) [inline src/modules.F90:299]: I inlined `init_usermass` for simplicity

File: configure [hunk 6]
```diff
@@ -43,12 +43,14 @@ OPTIONS
 "
 }
 
+warning_flags="-Wall -Wno-integer-division -Wno-maybe-uninitialized"
```
Comment (danielhollas) [inline configure:46]: To remove some spurious compiler warnings. We should now be warnings-free with this setting I think.

**Comment by danielhollas** [comment]
@suchanj thanks for looking at the other PR. :bowing_man:  This one is now ready as well, but absolutely no pressure, do on your own time. Thanks! This is a mixture of several small things, see first comment.

**Review by suchanj — APPROVED** [review]
Looking good!

**Pull request merged**